.azure-pipelines/generate-beta-typings.yml

```diff
@@ -0,0 +1,166 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# This pipeline contains steps to generate the typings from microsoft-graph meta-data using the TypeWriter. 
+name: $(BuildDefinitionName)_$(SourceBranchName)_$(Date:yyyyMMdd)$(Rev:.r)
+
+trigger: none # disable triggers based on commits.
+pr: none # disable triggers based on pull requests.
+
+resources:
+  repositories:
+  - repository: msgraph-typescript-typings # The name used to reference this repository in the checkout step
+    type: github
+    endpoint: microsoftgraph
+    name: microsoftgraph/msgraph-typescript-typings
+    ref: beta # checkout the beta branch
+  - repository: msgraph-metadata
+    type: github
+    endpoint: microsoftgraph
+    name: microsoftgraph/msgraph-metadata
+    ref: master #
+  pipelines:
+  - pipeline: publishMetadata # This pipeline validates and produces an metadata artifact that we use for generation.
+    source: (beta - 3) msgraph-publish-cleanmetadata
+    trigger:
+      branches:
+      - master
+
+pool:
+  vmImage: windows-latest # Info about this image: [0][1]
+
+variables:
+  - group: MicrosoftGraph # Variable group, where variables not set here come from. Set in Azure DevOps
+
+steps:
+- checkout: msgraph-typescript-typings
+  clean: true
+  fetchDepth: 1
+  persistCredentials: true
+- checkout: msgraph-metadata
+  clean: true
+  fetchDepth: 1
+
+- task: PowerShell@2 # Setup environment variables and make them available to all tasks. See [1] for more info.
+  displayName: 'Calculate and set pipeline variables for this job'
+  inputs:
+    targetType: inline
+    script: |
+      $repoModelsDir = "$env:Build_SourcesDirectory\msgraph-typescript-typings\"
+      Write-Host "Path to repo model directory: $repoModelsDir"
+      Write-Host "##vso[task.setvariable variable=repoModelsDir]$repoModelsDir"
+
+      $outputPath = Join-Path $env:Build_SourcesDirectory "output"
+      Write-Host "Path to typewriter.exe output $outputPath"
+      Write-Host "##vso[task.setvariable variable=outputPath]$outputPath"
+
+      $cleanMetadata = "https://raw.githubusercontent.com/microsoftgraph/msgraph-metadata/master/clean_beta_metadata/cleanMetadataWithDescriptionsbeta.xml"
+      Write-Host "Path to clean metadata $cleanMetadata"
+      Write-Host "##vso[task.setvariable variable=cleanMetadata]$cleanMetadata"
+
+      $branchName = "pipelinebuild/beta/$env:Build_BuildId" # Match the spec in the GH Action
+      Write-Host "Branch path spec for the pull request will be $branchName"
+      Write-Host "##vso[task.setvariable variable=branchName]$branchName"
+
+- task: PowerShell@2
+  displayName: 'Git: checkout beta'
+  inputs:
+    targetType: inline
+    workingDirectory: '$(Build.SourcesDirectory)/msgraph-typescript-typings'
+    script: |
+      git checkout beta | Write-Host
```

Comment:
No need for this step. Checkout step above handles it. This will prevent you from running the pipeline on feature branches without a code change. In other words, one-off manual runs have an option to change `beta` branch to your feature branch, but this step forces to checkout `beta`, which will override that manual setting.

Reply:
beta branch is like a master branch for any beta purpose. This step is important to branch of the beta branch or any beta version related tasks.

Reply:
Yes and it is already handled by the ref you define above.

Check this step:
https://o365exchange.visualstudio.com/O365%20Sandbox/_build/results?buildId=3671694&view=logs&j=12f1170f-54f2-53f3-20dd-22fc7dff55f9&t=f675c248-5d9b-5ae2-013f-523aba835bc2&l=407

`git checkout --progress --force 04bd155b713068532da364527085f3eb0e957c38`

`04bd155b713068532da364527085f3eb0e957c38` is what `beta` branch points to.

This is for the scenario where you fork from `beta`, make some changes and you want to run the pipeline from there. With the hardcoded checkout step to beta, you won't be able change that using Azure DevOps UI for manual runs.

Reply:
Got it. Thanks for the explanation! I removed that step. Reran the [pipeline ](https://o365exchange.visualstudio.com/O365%20Sandbox/_build/results?buildId=3672153&view=logs&j=12f1170f-54f2-53f3-20dd-22fc7dff55f9) without the checkout step and it works fine.